restapi/src/main/java/io/stargate/web/resources/v2/RowsResource.java

```diff
@@ -380,8 +382,14 @@ public Response createRow(
           AuthenticatedDB authenticatedDB =
               db.getRestDataStoreForToken(token, getAllHeaders(request));
 
-          @SuppressWarnings("unchecked")
-          Map<String, Object> requestBody = mapper.readValue(payload, Map.class);
+          Map<String, Object> requestBody;
+          try {
+            requestBody =
+                mapper.readValue(payload, new TypeReference<HashMap<String, Object>>() {});
+          } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(
+                String.format("Input provided is not valid json. %s", e.getMessage()));
```

Comment:
Aside from question of what to catch, what to throw, would it make sense to include type of exception caught here like:

```
String.format("(%s) Input provider not valid json. %s", e.getClass().getName(), e.getMessage());
```

just in case more investigation was needed? Or I guess alternatively use `e.toString()` which adds type as well (but also includes location information which may or may not be desirable).

Reply:
I'm not sure if the type of exception will be that valuable for API consumers. They'll be more concerned with the "invalid json" message.

Reply:
Right, I was more thinking of us (datastax admins, devs) looking at logs/failures, not end users (except if and when they need to relay that information to us).
Not sure if there is a way to retain more information for our consumption.

But it is probably true that error message itself is enough to backtrack into type if we need that on case-by-case basis (just cannot search for specific type).